examples/reminders/lib/base/data_sources/remote/objectbox_remote_data_source.dart

```diff
@@ -0,0 +1,208 @@
+      .build()
+      .find()
+      .length;
+
+  @override
+  Future<int> getIncompleteCount() async => (await _remindersBox)
+      .query(ObjectBoxCloudReminderModel_.complete.equals(false))
+      .build()
+      .find()
+      .length;
+
+  @override
+  Future<ReminderPair> update(ReminderModel updatedModel) async {
+    var oldModel = (await _remindersBox).get(int.parse(updatedModel.id));
+    if (oldModel == null) {
+      throw Exception('Old model not found');
+    }
+    final oldModelId = int.tryParse(updatedModel.id);
+    if (oldModelId == null) {
+      throw Exception('Old model id is not a number');
+    }
+    (await _remindersBox).put(updatedModel.toReminderObjectboxLocalModel(
+      oldModelId,
+      updatedModel.title,
+      updatedModel.dueDate,
+      updatedModel.complete,
+      updatedModel.authorId,
+    ));
+    return ReminderPair(
+      old: oldModel.toReminderModel(),
+      updated: updatedModel,
+    );
+  }
+
+  /// Returns the id of the logged in user or null if no user is logged in
+  Future<String?> _getAuthorIdOrNull() async {
+    var user = await _storage.read(key: _authorId);
+    if (user == _anonymous) {
+      return null;
+    }
+    return user;
+  }
+
+  Condition<ObjectBoxCloudReminderModel> getCondition(
+      ReminderModelRequest? request, String? authorId) {
+    Condition<ObjectBoxCloudReminderModel>? condition;
+    DateTime from;
+    DateTime to;
+
+    if (request?.filterByDueDateRange != null) {
+      from = request?.filterByDueDateRange?.from ?? DateTime.now();
+      to = request?.filterByDueDateRange?.to ?? DateTime.now();
+    } else {
+      from = DateTime.now().subtract(const Duration(days: 365));
+      to = DateTime.now().add(const Duration(days: 365));
+    }
+    condition = ((authorId == null
+        ? ObjectBoxCloudReminderModel_.authorId.isNull()
+        : ObjectBoxCloudReminderModel_.authorId.equals(authorId)));
+    if (request?.complete != null) {
+      condition.and(
+          ObjectBoxCloudReminderModel_.complete.equals(request!.complete!));
+    }
+
+    condition.and(ObjectBoxCloudReminderModel_.dueDate.betweenDate(from, to));
+    return condition;
+  }
+
+  int getOrder(ReminderModelRequestSort? sort) {
+    int order = 0;
+    if (sort != null) {
+      switch (sort) {
+        case ReminderModelRequestSort.dueDateAsc:
+          break;
+        case ReminderModelRequestSort.dueDateDesc:
+          order = Order.descending;
+      }
+    }
+    return order;
+  }
+}
+
+extension ReminderModelMapper on ReminderModel {
+  ObjectBoxCloudReminderModel toReminderObjectboxLocalModel(
+      int id, String title, DateTime dueDate, bool complete, String? authorId) {
+    return ObjectBoxCloudReminderModel(
+      id: id,
+      title: title,
+      dueDate: dueDate,
+      complete: complete,
+      authorId: authorId,
+    );
+  }
+}
+
+extension ReminderRealmModelMapper on ObjectBoxCloudReminderModel {
```

Comment:
Do we really need to have a separate data source and DTOs for the cloud ?

Reply:
Yes, DTOs have different annotations. Cloud uses `@Sync` annotation allowing it to connect to the sync server, but in order to use this DTO, it is required to initialize the sync client.
If we try to use DTO with `@Sync` annotation with local storage, an error is thrown that the sync client is not initialized.